### DISCUSSION TITLE
Large skin import causes the graphic thread to freeze

### DISCUSSION THREAD

**Discussion by Philosoph228** [ppy/osu#20813] [Q&A]
### Type

Performance

### Bug description

I have tried to install this skin
https://osu.ppy.sh/community/forums/topics/839786
It weights 137MB and contain lots of sequenced images for animations

The osu! deadly hanged for about a minute, a window title bar appeared on top in full screen mode, but music continued to play.
It seems that osu was about to crash. There was no input handing, nor screen updates.

I think it could be better to make skin importing process flawlessly asynchronous or something.

### Screenshots or videos

![osu!_vskMSGdYEE](https://user-images.githubusercontent.com/13369540/196495610-09f0aa5e-e373-4f8f-9f6e-b27c9c2e2b4a.png)

### Version

osu! 2022.1008.2-lazer

### Logs

Can't locate osu!lazer logs folder. It seems there none.

**Comment by Cesan** [comment]
I can't reproduce this on my end. Also lazer always produces logs, there is a "logs" folder where the realm is located. For quick access to this folder you can use the "open osu! folder" in the general settings.

**Comment by peppy** [comment]
Please give this another try after the next release. I have a feeling that #20712 will help.
